modules/debug_cmd/debug_cmd.c

```diff
@@ -82,6 +82,36 @@ static int cmd_ua_debug(struct re_printf *pf, void *unused)
 }
 
 
+/**
+ * Returns the current User-Agent and general codec state.
+ * Formatted as JSON, for use with TCP / MQTT API interface.
+ *
+ * @return Current User-Agent, NULL if none
+ */
+static int cmd_api_uastate(struct re_printf *pf, void *unused)
+{
+	(void)unused;
+	int err;
+	struct odict *od = NULL;
+
+	err = odict_alloc(&od, 8);
+	if (err)
+		return err;
+
+	struct le *le;
+	for (le = list_head(uag_list()); le && !err; le = le->next) {
+		const struct ua *ua = le->data;
+
+		err |= ua_state_json_api(od, pf, ua);
+		err |= json_encode_odict(pf, od);
+
+		if (err)
+			warning("debug: failed to encode json (%m)\n", err);
+	}
```

Comment:
please destroy the object allocated in line 97:

```c
mem_deref(od);
```

Reply:
Ah, seems important, as mentioned not so familiar with how to deal with the "cleanup". Adding it after the for-loop.